contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,751 @@
+    )
+        external
+        onlyAdminACL(this.addDependencyTypeAdditionalRepository.selector)
+        onlyNonEmptyString(_additionalRepository)
+    {
+        uint24 additionalRepositoryCount = dependencyTypeInfo[_dependencyTypeId]
+            .additionalRepositoryCount;
+        dependencyTypeInfo[_dependencyTypeId].additionalRepositories[
+                additionalRepositoryCount
+            ] = _additionalRepository;
+        dependencyTypeInfo[_dependencyTypeId].additionalRepositoryCount =
+            additionalRepositoryCount +
+            1;
+
+        emit DependencyTypeAdditionalRepositoryUpdated(
+            _dependencyTypeId,
+            _additionalRepository,
+            additionalRepositoryCount
+        );
+    }
+
+    /**
+     * @notice Removes additional repository for depenency `_dependencyId` at index `_index`.
+     * Removal is done by swapping the element to be removed with the last element in the array, then deleting this last element.
+     * Assets with indices higher than `_index` can have their indices adjusted as a result of this operation.
+     * @param _dependencyTypeId Dependency type to be updated.
+     * @param _index Additional repository index.
+     */
+    function removeDependencyTypeAdditionalRepositoryAtIndex(
+        bytes32 _dependencyTypeId,
+        uint256 _index
+    )
+        external
+        onlyAdminACL(
+            this.removeDependencyTypeAdditionalRepositoryAtIndex.selector
+        )
+        onlyExistingDependencyType(_dependencyTypeId)
+    {
+        uint24 additionalRepositoryCount = dependencyTypeInfo[_dependencyTypeId]
+            .additionalRepositoryCount;
+        require(_index < additionalRepositoryCount, "Asset index out of range");
+
+        uint24 lastElementIndex = additionalRepositoryCount - 1;
+
+        dependencyTypeInfo[_dependencyTypeId].additionalRepositories[
+                _index
+            ] = dependencyTypeInfo[_dependencyTypeId].additionalRepositories[
+            lastElementIndex
+        ];
+        delete dependencyTypeInfo[_dependencyTypeId].additionalRepositories[
+            lastElementIndex
+        ];
+
+        dependencyTypeInfo[_dependencyTypeId]
+            .additionalRepositoryCount = lastElementIndex;
+
+        emit DependencyTypeAdditionalRepositoryRemoved(
+            _dependencyTypeId,
+            _index
+        );
+    }
+
+    /**
+     * @notice Updates additional repository for dependency type `_dependencyTypeId` at `_index`.
+     * @param _dependencyTypeId Dependency type to be updated.
+     * @param _index Additional repository index.
+     * @param _additionalRepository New Repository URL.
+     */
+    function updateDependencyTypeAdditionalRepositoryAtIndex(
+        bytes32 _dependencyTypeId,
+        uint256 _index,
+        string memory _additionalRepository
+    )
+        external
+        onlyAdminACL(
+            this.updateDependencyTypeAdditionalRepositoryAtIndex.selector
+        )
```

Comment:
maybe a no empty string modifier?

Reply:
In this case agree re: no-empty-string modifier.